src/core/buffers/representation/get_needed_segments.ts

```diff
@@ -172,6 +178,13 @@ function shouldContentBeReplaced(
     return true; // replace segments from another Adaptation
```

Comment:
Wait, isn't there still a problem if this condition is true?

Reply:
I think so